---
title: "Genomics Data Repositories"
author: "Rockefeller University, Bioinformatics Resource Centre"
date: "http://rockefelleruniversity.github.io/Plotting_In_R/"
output: 
  xaringan::moon_reader:
    css: ["default", "metropolisCustom.css", "metropolis-fontsCustom.css"]
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
  html_document:
    toc: true # table of content true
    toc_float: yes
    depth: 3  # upto three depths of headings (specified by #, ## and ###)
    number_sections: false  ## if you want number sections at each table header
    theme: united  # many options for theme, this one is my favorite.
    highlight: tango  # specifies the syntax highlighting style



---


```{r setup_ggplot2, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
AsSlides <- TRUE
```

# Data Repositories

## Getting hold of HTS data
- From public repositories.
- From collaborators.
- By sequencing some of your own material!



---
class: inverse, center, middle

# Repositories for HTS
<html><div style='float:left'></div><hr color='#EB811B' size=1px width=720px></html> 

---
## Public Repositories for HTS

- Several public sources of HTS data exist.
- First concentrating on those acting as repositories.
    - GEO (Gene Expression Omnibus).
    - ENA (European Nucleotide Database).
    - SRA (Short Read Archive).


---

## Gene Expression Omnibus

- GEO (https://www.ncbi.nlm.nih.gov/geo/)

.pull-left[
<img src="imgs/geo.png" alt="igv" height="300" width="400">
  ]
.pull-right[
- GEO holds different types of biological datasets.
- Very popular for submission of data accompanying publication.
- Captures metadata,  processed files and raw data.
- GEO was not built for HTS data.
  ]


---

## Gene Expression Omnibus

- GEO (https://www.ncbi.nlm.nih.gov/geo/)

<iframe width="100%" height= "80%" src="https://www.ncbi.nlm.nih.gov/geo/"></iframe>

---

## Short Read Archive

- SRA (www.ncbi.nlm.nih.gov/sra)

.pull-left[
<img src="imgs/sra.png" alt="igv" height="300" width="400">
  ]
.pull-right[
- NCBI's HTS specific repository.
- Sequencing specific metadata.
- Stores Raw data (in SRA format)
- SRA format - requires SRA Toolkit
  ]


---

## Short Read Archive

- SRA (www.ncbi.nlm.nih.gov/sra)

<iframe width="100%" height= "80%" src="https://www.ncbi.nlm.nih.gov/sra"></iframe>

---

## European Nucleotide Archive

- ENA (https://www.ebi.ac.uk/ena)

.pull-left[
<img src="imgs/ena.png" alt="igv" height="300" width="400">
  ]
.pull-right[
- ENA acts as a european HTS repository.
- Mirrors much of SRA.
- Stores Raw data
- No SRA formats - fastq by default.
  ]


---

## European Nucleotide Archive

- ENA (https://www.ebi.ac.uk/ena)

<iframe width="100%" height= "80%" src="https://www.ebi.ac.uk/ena"></iframe>


---

## Other Repositories
.pull-left[
<img src="imgs/encode.png" alt="igv" height="500" width="400">

]
.pull-right[
- Many repositories contain processed or unprocessed data.
- These typically are the result or a consortium's data release policies.
- Good example is Encode site. 
    - (https://www.encodeproject.org/)
- UCSC has many useful links to genomics data in various formats.
    - (http://hgdownload.soe.ucsc.edu/downloads.html)
]

---

## Encode Portal

Encode portal provides access to raw and processed/standardised results.

<iframe width="100%" height= "80%" src="https://www.encodeproject.org"></iframe>

---

## Repositories for processed data

.pull-left[
<img src="imgs/recount2.png" alt="igv" height="300" width="400">
<img src="imgs/ExprAtlas.png" alt="igv" height="300" width="400">

]
.pull-right[
- Other specialist repositories exist.
- [ReCount2](https://jhubiostatistics.shinyapps.io/recount/) database provides standardised counts for user analysis.
- Other databases like Immgen/Bodymap/[expression atlas](https://www.ebi.ac.uk/gxa/home) provide RNAseq for specific cells/tissues.
]

---

##  Reference data
- Reference Genome available from many locations.
- Different assemblies
    - Major Revisisons - Change locations
    - Minor Revisions - Update annotation
- Genome sequence stored as FASTA.
- Gene build as GFF3 or GTF.
- [IGenomes](https://support.illumina.com/sequencing/sequencing_software/igenome.html) contains full annotation files for many genomes.

