<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Genomics Data Repositories</title>
    <meta charset="utf-8" />
    <meta name="author" content="Rockefeller University, Bioinformatics Resource Centre" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="metropolisCustom.css" type="text/css" />
    <link rel="stylesheet" href="metropolis-fontsCustom.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Genomics Data Repositories
### Rockefeller University, Bioinformatics Resource Centre
### <a href="http://rockefelleruniversity.github.io/Plotting_In_R/" class="uri">http://rockefelleruniversity.github.io/Plotting_In_R/</a>

---






## Getting hold of HTS data
- From public repositories
- From collaborators
- By sequencing some of your own material!



---
class: inverse, center, middle

# Repositories for HTS
&lt;html&gt;&lt;div style='float:left'&gt;&lt;/div&gt;&lt;hr color='#EB811B' size=1px width=720px&gt;&lt;/html&gt; 

---
## Public Repositories for HTS

- Several public sources of HTS data exist.
- First concentrating on those acting as repositories.
    - GEO (Gene Expression Omnibus)
    - ENA (European Nucleotide Database)
    - SRA (Short Read Archive)


---

## Gene Expression Omnibus

- GEO (https://www.ncbi.nlm.nih.gov/geo/)

.pull-left[
&lt;img src="imgs/geo.png" alt="igv" height="300" width="400"&gt;
]
.pull-right[
- GEO holds different types of biological datasets.
- Very popular for submission of data accompanying publication.
- Captures metadata,  processed files and raw data.
- GEO was not built for HTS data 
]


---

## Gene Expression Omnibus

- GEO (https://www.ncbi.nlm.nih.gov/geo/)

&lt;iframe width="100%" height= "80%" src="https://www.ncbi.nlm.nih.gov/geo/"&gt;&lt;/iframe&gt;

---

## Short Read Archive

- SRA (www.ncbi.nlm.nih.gov/sra)

.pull-left[
&lt;img src="imgs/sra.png" alt="igv" height="300" width="400"&gt;
]
.pull-right[
- NCBI's HTS specific repository.
- Sequencing specific metadata.
- Stores Raw data (in SRA format)
- SRA format - requires SRA Toolkit
]


---

## Short Read Archive

- SRA (www.ncbi.nlm.nih.gov/sra)

&lt;iframe width="100%" height= "80%" src="https://www.ncbi.nlm.nih.gov/sra"&gt;&lt;/iframe&gt;

---

## European Nucleotide Archive

- ENA (https://www.ebi.ac.uk/ena)

.pull-left[
&lt;img src="imgs/ena.png" alt="igv" height="300" width="400"&gt;
]
.pull-right[
- ENA acts as a european HTS repository.
- Mirrors much of SRA.
- Stores Raw data
- No SRA formats - fastq by default.
]


---

## European Nucleotide Archive

- ENA (https://www.ebi.ac.uk/ena)


&lt;iframe width="100%" height= "80%" src="https://www.ebi.ac.uk/ena"&gt;&lt;/iframe&gt;


---

## Other Repositories
.pull-left[
&lt;img src="imgs/encode.png" alt="igv" height="500" width="400"&gt;

]
.pull-right[
- Many repositories contain processed or unprocessed data.
- These typically are the result or a consortium's data release policies.
- Good example is Encode site. 
    - (https://www.encodeproject.org/)
- UCSC has many useful links to genomics data in various formats.
    - (http://hgdownload.soe.ucsc.edu/downloads.html)
]

---

## Encode Portal

Encode portal provides access to raw and processed/standardised results.

&lt;iframe width="100%" height= "80%" src="https://www.encodeproject.org"&gt;&lt;/iframe&gt;

---

## Repositories for processed data

.pull-left[
&lt;img src="imgs/recount2.png" alt="igv" height="300" width="400"&gt;
&lt;img src="imgs/ExprAtlas.png" alt="igv" height="300" width="400"&gt;

]
.pull-right[
- Other specialist repositories exist.
- [ReCount2](https://jhubiostatistics.shinyapps.io/recount/) database provides standardised counts for user analysis.
- Other databases like Immgen/Bodymap/[expression atlas](https://www.ebi.ac.uk/gxa/home) provide RNAseq for specific cells/tissues.
]

---

##  Reference data
- Reference Genome available from many locations.
- Different assemblies
    - Major Revisisons - Change locations
    - Minor Revisions - Update annotation
- Genome sequence stored as FASTA.
- Gene build as GFF3 or GTF.
- [IGenomes](https://support.illumina.com/sequencing/sequencing_software/igenome.html) contains full annotation files for many genomes.
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  /* Replace <script> tags in slides area to make them executable
   *
   * Runs after post-processing of markdown source into slides and replaces only
   * <script>s on the last slide of continued slides using the .has-continuation
   * class added by xaringan. Finally, any <script>s in the slides area that
   * aren't executed are commented out.
   */
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container:not(.has-continuation) script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
  var scriptsNotExecuted = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container.has-continuation script'
  );
  if (!scriptsNotExecuted.length) return;
  for (var i = 0; i < scriptsNotExecuted.length; i++) {
    var comment = document.createComment(scriptsNotExecuted[i].outerHTML)
    scriptsNotExecuted[i].parentElement.replaceChild(comment, scriptsNotExecuted[i])
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
